### PR TITLE
Add Region, AccessType and Deployment End Time metrics to Telemetry

### PR DESCRIPTION
**Pull request by Viraj27**
# Description

This PR is for adding the following metrics to the telemetry payload:
1. Region/Location of deployment.
2. AccessType (Public/Private)
3. Deployment End Time (RTC3339 / ISO 8601 format)

## PR task checklist

- [x] I have performed a self-review of my code
- [x] I have followed code style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~[ ] I have made corresponding changes to the documentation~
- ~[] I have added thorough tests~
- [x] I have had a successful deployment with the changes in this PR

# Ticket

https://issues.redhat.com/browse/AAP-10508
https://issues.redhat.com/browse/AAP-10507
https://issues.redhat.com/browse/AAP-10482

# Testing

1. Pull down the PR.
2. Create a new deployment using the installerincluded branch of aap-azurerm repo.
3. Once the Deployment is complete, go to Segment (https://app.segment.com/redhat-devtools/sources/ansible_on_azure_managed_app_dev/debugger) and check if the event is published here.
4. Check for the properties published in the PR.

### CODE REVIEW

**Comment by tznamena** [comment]
First, let's move the Jira items out of the subject/title of the PR and into the description. The title should describe, at a high level, what the PR delivers.

File: server/engine/telemetry.go [hunk 1]
```diff
@@ -34,6 +35,21 @@ func BuildSegmentPropertiesMap(db *gorm.DB) analytics.Properties {
...
+	//var outputsMap map[string]interface{}
+	var outputs []model.Output
+	db.Find(&outputs)
+	for _, data := range outputs {
+		if data.ModuleName == "" {
+			location := data.Values["location"].(map[string]interface{})["value"]
+			accessType := data.Values["access"].(map[string]interface{})["value"]
+			model.SetMetric(db, model.Region, location.(string))
+			model.SetMetric(db, model.AccessType, accessType.(string))
+		}
```
Comment (bhavenst) [inline server/engine/telemetry.go:49]: You can add a break so you don't keep cycling through the rest of the outputs..

**Review by bhavenst — CHANGES_REQUESTED** [review]

File: server/engine/telemetry.go [hunk 2]
```diff
@@ -34,6 +35,21 @@ func BuildSegmentPropertiesMap(db *gorm.DB) analytics.Properties {
 	return propertiesMap
 }
 
+func GetMetricFromMainOutputs(db *gorm.DB) {
```
Comment (tznamena) [inline server/engine/telemetry.go:38]: Functions starting with `Get...` are known as getters and they should return something. In this case we are storing metrics so we should name it something like `StoreMetricsFromMainOutputs`.

File: server/engine/telemetry.go [hunk 3]
```diff
@@ -34,6 +35,21 @@ func BuildSegmentPropertiesMap(db *gorm.DB) analytics.Properties {
...
+
+	//var outputsMap map[string]interface{}
+	var outputs []model.Output
+	db.Find(&outputs)
+	for _, data := range outputs {
+		if data.ModuleName == "" {
+			location := data.Values["location"].(map[string]interface{})["value"]
```
Comment (tznamena) [inline server/engine/telemetry.go:45]: This block needs a couple of checks to ensure we don't panic when data does not contain `location` and/or `access`.

File: server/engine/telemetry.go [hunk 4]
```diff
@@ -34,6 +35,24 @@ func BuildSegmentPropertiesMap(db *gorm.DB) analytics.Properties {
 	return propertiesMap
 }
 
+func StoreMetricFromMainOutputs(db *gorm.DB) {
+
+	//var outputsMap map[string]interface{}
```
Comment (bhavenst) [inline server/engine/telemetry.go:40]: remove comment?

File: server/engine/telemetry.go [hunk 5]
```diff
@@ -34,6 +35,24 @@ func BuildSegmentPropertiesMap(db *gorm.DB) analytics.Properties {
...
+	var mainOutput model.Output
+	db.Where("module_name = ?", "").Find(&mainOutput)
+
+	if loc, exists := mainOutput.Values["location"]; exists {
+		model.SetMetric(db, model.Region, loc.(map[string]interface{})["value"].(string))
+	} else {
+		log.Errorf("Location of deployment is missing : will not be included in telemetry")
```
Comment (bhavenst) [inline server/engine/telemetry.go:47]: Should be log.Error w/ no formatting template included.

File: server/engine/telemetry.go [hunk 6]
```diff
@@ -34,6 +35,24 @@ func BuildSegmentPropertiesMap(db *gorm.DB) analytics.Properties {
...
+	}
+	if access, exists := mainOutput.Values["access"]; exists {
+		model.SetMetric(db, model.AccessType, access.(map[string]interface{})["value"].(string))
+	} else {
+		log.Errorf("Access Type of deployment is missing : will not be included in telemetry")
```
Comment (bhavenst) [inline server/engine/telemetry.go:52]: ditto

**Review by bhavenst — COMMENTED** [review]
Couple of minor things..

File: server/engine/telemetry.go (hunk 5 above)
Comment (Viraj27) [reply server/engine/telemetry.go]: Updated

File: server/engine/telemetry.go (hunk 4 above)
Comment (Viraj27) [reply server/engine/telemetry.go]: Updated

File: server/engine/telemetry.go (hunk 6 above)
Comment (Viraj27) [reply server/engine/telemetry.go]: Updated

File: server/engine/telemetry.go (hunk 3 above)
Comment (Viraj27) [reply server/engine/telemetry.go]: Added if checks for region and accessType

File: server/engine/telemetry.go (hunk 2 above)
Comment (Viraj27) [reply server/engine/telemetry.go]: Changed it to `StoreMetricsFromMainOutputs`

**Review by bhavenst — APPROVED** [review]

**Review by tznamena — APPROVED** [review]

**Pull request merged**